src/Sylius/Bundle/RbacBundle/spec/Form/EventSubscriber/AddParentFormSubscriberSpec.php

```
@@ -0,0 +1,81 @@
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class AddParentFormSubscriberSpec extends ObjectBehavior
+{
+    function let()
+    {
+       $this->beConstructedWith('role');
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType(AddParentFormSubscriber::class);
+    }
+
+    function it_implements_event_subscriber_interface()
+    {
+        $this->shouldImplement(EventSubscriberInterface::class);
+    }
+
+    function it_subscribes_to_event()
+    {
+        $this::getSubscribedEvents()->shouldReturn(array(FormEvents::PRE_SET_DATA => 'preSetData'));
+    }
+
+    function it_add_parent_if_it_is_not_set_and_resource_has_not_id(FormEvent $event, FormInterface $form, RoleInterface $role)
+    {
+        $event->getData()->willReturn($role);
+        $event->getForm()->willReturn($form);
+
+        $role->getId()->willReturn(null);
+        $role->getParent()->willReturn(null);
+
+        $form->add('parent', 'sylius_role_choice', Argument::cetera())->shouldBeCalled();
+
+        $this->preSetData($event);
+    }
+
+    function it_add_parent_if_it_is_set_and_the_resource_has_id(FormEvent $event, FormInterface $form, RoleInterface $role)
+    {
+        $event->getData()->willReturn($role);
+        $event->getForm()->willReturn($form);
+
+        $role->getId()->willReturn(112);
+        $role->getParent()->willReturn(Argument::type(RoleInterface::class));
+
+        $form->add('parent', 'sylius_role_choice', Argument::cetera())->shouldBeCalled();
+
+        $this->preSetData($event);
+    }
+
+    function it_throws_exception_when_resource_does_not_implement_role_interface_or_permission_interface(FormEvent $event, FormInterface $form, $resource)
+    {
+        $event->getData()->willReturn($resource);
+        $event->getForm()->willReturn($form);
+
+        $this->shouldThrow('\UnexpectedTypeException');
```

Does it work? I think it should throw ``'Sylius\Bundle\RbacBundle\Form\Exception\UnexpectedTypeException'``.